src/index.js

```diff
@@ -284,10 +284,49 @@ const wealthsimple = {
    * Information about a security on the WealthSimple Trade Platform.
    *
    * @param {*} tokens The access and refresh tokens returned by a successful login.
-   * @param {*} ticker The security symbol
+   * @param {string|object} ticker The security symbol. An exchange may be added as a suffix, separated from the symbol with a colon.
+   * @param {string} ticker.symbol The security symbol. An exchange may be added as a suffix, separated from the symbol with a colon.
+   *                               If an exchange suffix is present, then ticker.exchange may only be passed if it references the same exchange.
+   * @param {string} [ticker.exchange] (optional) the exchange the security trades in
+   * @param {string} [ticker.id] (optional) The internal WealthSimple Trade security ID
    */
-  getSecurity: async (tokens, ticker) =>
-    handleRequest(endpoints.SECURITY, { ticker }, tokens),
+  getSecurity: async (tokens, ticker) => {
```

Comment:
I like the approach of allowing the user to either pass a string that may have the exchanged postfixed, or passing an object with 3 separate keys. However, I do think it's unnecessary that the symbol key in the object format still allows for an exchange to be postfixed. If you are choosing the object format as a user, you are either doing so to (i) specify an internal id; or (ii) to specify the symbol and exchange separately. If you wanted to specify the symbol and exchange using the colon format, you would simply pass a string rather than use the object format. With this reasoning, i think it would be simpler to just specify that the symbol key in the ticker object would simply be the symbol of the security sans any exchange postfixes.

Reply:
I mean it's still supported. It just rejects the promise if you condradict yourself.

Reply:
I still believe this could be introducing confusion into the API for the user. This would essentially assert that our usage of `ticker` and `symbol` are interchangeable, which is honestly confusing. I think the intent behind the expansion of the ticker into a 3-key object is that a ticker can be a composite of the security symbol and the exchange suffix. Supporting the symbol to also be a fully qualified ticker would be, in my opinion, unproductive. This would also be introducing an unnecessary failure point that you have to cover to reject the promise.

I would appreciate it if you could write out why you believe the current implementation makes more sense than what i am suggesting. I really want us to get on the same page, but honestly, from where this stands, I do not want to support things for the sake of supporting them without value added to the users of the API.

Reply:
The implementation here makes more sense to me becuase of the following use case:
- variable `ticker` that may or may not be exchange-suffixed
- variable `exchange` that may or may not be `null`
- You pass `{symbol: ticker, exchange: exchange}` in and do not have to worry about anything. If `exchange` is `null` then the code will fall-back to the possible exchange information in the ticker string. If that is also absent and there is only one result from the query, then you get that. If there is more than one possible result, the promise is rejected and protects you. If the ticker string is exchange-suffixed and you also pass an exchange, then you don't need to worry about anything so long as they agree. If they don't agree then the promise rejects and protects you.

If you are forbidden from exchange-suffixing a ticker string when using the object form, then the calling code has to explicitly check if their ticker string is exchange-suffixed and remove it in this situation, because the WealthSimple API will not produce any results for an exchange-suffixed query.

In my opinion this makes the wrapper more robust and user-friendly, however I understand your wish to keep it simple, so I've updated the PR to remove the functionality.

Reply:
Thank you for taking the time to point out why you think your original approach is a better one. I understand that your concern was coming from trying to make the API as failproof as possible, which is definitely a legitimate viewpoint regarding developing APIs for public consumption.

I do think that while the intent was noble, I feel like it was trying to protect the user from any failures at the expense of masking logic errors on the user's side. If someone is using the object notation to explicitly specify the `exchange` and somehow their workflow turns up to provide a null`exchange`, that means something has gone wrong on their side. With your original proposed implementation, this error might be masked from the API users which might cause them grief in figuring out their logical errors.

Which brings me to the next point: how would it be possible that the user can suffix the exchange in the `symbol` element, but not the `exchange` element? Their source would have to be the same. I do not think there should ever be a logical flow were `exchange` is suddenly null while the exchange postfix in the `symbol` element is still present. If the user is able to postfix the exchange in the symbol, they should be able to add it in the `exchange` element of the object.

Here is an example of how i expect someone to consume your proposed API

```javascript

let exchange = "TSE";

getSecurity(tokens, {
    symbol: `BB:${exchange}`,
    exchange: exchange
}.then(data => console.log(data));
```

For your scenario to happen where `exchange` goes null, someone would have to do the following:

```javascript

let exchange = "TSE";
let otherExchangeSource = "TSE"; // a separate source of truth for exchange that could become null

getSecurity(tokens, {
    symbol: `BB:${exchange}`,
    exchange: otherExchangeSource
}.then(data => console.log(data));
```

And this is what I am stuck on: why would anyone attempt to pass in two separate sources of truth for the exchange?

--

The way i proposed should have behave as follows:

* You want to keep it simple and pass one string with a suffixed exchange? `BB:TSE` - you're good to go.
* You prefer to split up the ticker into a symbol and exchange? no problem - `{symbol: "BB", exchange: "TSE"}`, we got you covered.